src/variables.jl

```diff
@@ -257,7 +257,9 @@ function has_lower_bound(v::VariableRef)
 end
 
 function lower_bound_index(v::VariableRef)
-    @assert has_lower_bound(v) # TODO error message
+    if !has_lower_bound(v)
+        error("Variable $(v) does not have a lower bound.")
```

Comment:
This might be misleading. Consider a polyhedral feasible set, if you eliminate all variables except `v` then it might be a set that is bounded from below.
Maybe `lower bound constraint` ?

Reply:
This error message will almost always be called when users run `JuMP.lower_bound(x)` if `x` is missing a lower bound.

There might be other constraints that enforce a lower bound on the variable, but that is a different kettle of fish.

Reply:
> that is a different kettle of fish.

That is clear for us, but is that clear for a user that has just started using JuMP and has no idea what `JuMP.lower_bound` is ?

Reply:
http://www.juliaopt.org/JuMP.jl/latest/variables/#Variable-bounds-1

Reply:
Who reads the doc if it is not thrown on your face :-P

Reply:
Also a user that has done `@constraint model x >= 0` will not understand why there is no lower bound.

Reply:
This relates to https://github.com/JuliaOpt/JuMP.jl/issues/1586. I feel that if the user writes `@constraint`, we should add a constraint (e.g., a row) to the problem. 

Then we make "variable bound" mean a bound that was introduced in `@variable` or via `JuMP.set_lower_bound` and `JuMP.set_upper_bound`.

Reply:
I think the error message is sufficiently clear given that `has_lower_bound` is a documented concept in JuMP.